packages/eslint-plugin/src/rules/no-unnecessary-type-constraint.ts

```diff
@@ -60,14 +65,22 @@ export default util.createRule({
         ])
       : new Map([[AST_NODE_TYPES.TSUnknownKeyword, 'unknown']]);
 
-    const inJsx = context.getFilename().toLowerCase().endsWith('tsx');
+    const filename = context.getFilename();
+    const scriptKind = getScriptKind(
+      filename,
+      !!context.parserOptions.ecmaFeatures?.jsx,
+    );
+    const inJsx = getLanguageVariant(scriptKind) === ts.LanguageVariant.JSX;
+    const nodeNextModuleExtension = isNodeNextModuleExtension(filename);
     const source = context.getSourceCode();
 
     const checkNode = (
       node: TypeParameterWithConstraint,
       inArrowFunction: boolean,
     ): void => {
-      const constraint = unnecessaryConstraints.get(node.constraint.type);
+      const constraint =
+        !nodeNextModuleExtension &&
+        unnecessaryConstraints.get(node.constraint.type);
```

Comment:
note that if we're in a `.mts` or `.cts` file - we still want to report and apply a fix!
It just needs to be the JSX-style fix, not the non-JSX fix.
I.e. instead of fixing `<T extends any>` to `<T>`, we need to fix it to `<T,>`.

Reply:
Is this correct though? In my experience, something like `prettier` will then format `<T,>` to `<T>` as also highlighted [here](https://github.com/typescript-eslint/typescript-eslint/issues/5449#issuecomment-1265318049). If you have a pipeline which employs `eslint` **and** `prettier` fixes, then the behaviour will be unexpected.